chain/src/main/java/org/ethereum/beacon/chain/observer/ObservableStateProcessorImpl.java

```diff
@@ -103,29 +105,22 @@ public ObservableStateProcessorImpl(
 
   @Override
   public void start() {
-    Flux.from(slotTicker).doOnNext(this::onNewSlot).subscribe();
-    Flux.from(attestationPublisher).doOnNext(this::onNewAttestation).subscribe();
-    Flux.from(beaconPublisher).doOnNext(this::onNewBlockTuple).subscribe();
-    this.regularJobExecutor =
-        Executors.newSingleThreadScheduledExecutor(
-            runnable -> {
-              Thread t = new Thread(runnable, "observable-state-processor-regular");
-              t.setDaemon(true);
-              return t;
-            });
-    regularJobExecutor.scheduleAtFixedRate(
-        this::doHardWork, 0, UPDATE_MILLIS, TimeUnit.MILLISECONDS);
-    this.continuousJobExecutor =
-        Executors.newSingleThreadScheduledExecutor(
-            runnable -> {
-              Thread t = new Thread(runnable, "observable-state-processor-continuous");
-              t.setDaemon(true);
-              return t;
-            });
+    Flux.from(slotTicker).subscribe(this::onNewSlot);
+    Flux.from(attestationPublisher).subscribe(this::onNewAttestation);
+    Flux.from(beaconPublisher).subscribe(this::onNewBlockTuple);
+    regularJobExecutor.executeAtFixedRate(
+        Duration.ZERO, Duration.ofMillis(UPDATE_MILLIS), this::doHardWork);
   }
 
   private void runTaskInSeparateThread(Runnable task) {
-    continuousJobExecutor.execute(task);
+    continuousJobExecutor.execute(() -> {
```

Comment:
maybe add `execute(Runnable task, Consumer<Exception> failCallback)`?

Reply:
This is just a temp solution. You may always handle returned `CompletableFuture`, but I'm going to make generic solution for all executors to log unhandled exceptions